### ISSUE TITLE
UNT0006 misidentifies static methods as Unity events

### ISSUE THREAD

**Issue by toxikman** [microsoft/Microsoft.Unity.Analyzers#182]
#### Bug description
UNT0006 misidentifies static methods with the same name as Unity events.

- Version of analyzers assembly: 1.11.1
- Analyzer rule: UNT0006
- Error (exception message, type, and callstack where applicable): 
info UNT0006: The Unity message 'Start' has an incorrect signature.

#### To Reproduce
```
using UnityEngine;
public class CoroutineManagerSingleton : MonoBehaviour
{
    static CoroutineManagerSingleton pInstance;

    void Awake()
    {
        Assert.IsFalse(pInstance);
        pInstance = this;
    }

    public static Coroutine Start(IEnumerator coroutine)
    {
        return pInstance.StartCoroutine(coroutine);
    }
}

```
#### Expected behavior
Static methods that have the same name as an event should not trigger this warning since static methods can never be Unity events.

**Comment by sailro** [comment]
I understand your point.

Technically Unity is able to use 'static' messages as well, we have an example in a recent PR:
https://github.com/microsoft/Microsoft.Unity.Analyzers/pull/180/files#diff-fcac62df312f8276e13cb9f50eeb6d02c9069711420b727040ca5e92d2291c71R730

Let's merge this one first, and we'll think about this issue then.

**Comment by toxikman** [comment]
I see, well maybe I should have said that this Start() event and others can never be static (not all Unity events). Perhaps there are instances where Unity looks for static events, but I believe Awake, Start, OnEnable, OnDisable, OnDestroy and all the standard ones cannot be static. I recognize that this isn't particular good practice anyway to have a method using this name, static or not. This warning was triggered for us in 3rd-party code.
